### DISCUSSION TITLE
Does the dev server include cookies?

### DISCUSSION THREAD

**Discussion by lockinmarv** [frandiox/vite-ssr#49] [Q&A] [answered]
Hey!

When calling my graphql endpoint from the client-side, the server prints my cookies but not when I call it from the server-side (e.g. onServerPrefetch or when checking `import.meta.env.SSR` in the entry function).
Is this because of `node-fetch` which doesn't include the credentials?

**Comment by frandiox** [comment, marked as answer]
Hello! Yeah there are differences on browser's fetch vs Node's fetch. The browser adds cookies automatically but of course Node doesn't since there are no cookies in that environment.

One thing you could do is access the original `request` object that is passed to the main hook and read the cookies from there. Then, you can provide them to your onServerPrefetch hooks via provide/inject or any other way. I think cookies can be added to node-fetch like this: https://stackoverflow.com/questions/34815845/how-to-send-cookies-with-node-fetch


Or perhaps  you can play with this idea:

```js
const originalFetch = globalThis.fetch

export default viteSSR(App, { routes }, ({ request }) => {
  if (import.meta.SSR) {
    globalThis.fetch = (url, options) => {
      return originalFetch(url, { ...options, cookie: request.cookie })
    }
  }
})
```

Perhaps this is something we could do at `vite-ssr` level if it's useful for most users 🤔

**Reply by lockinmarv** [reply]
I will try that, thanks for your help and the code! 😊
